src/bp/ui-studio/src/web/views/FlowBuilder/diagram/nodes/LinkWidget.tsx

```diff
@@ -1,29 +1,70 @@
+import _ from 'lodash'
 import React from 'react'
-import { AbstractLinkFactory, DefaultLinkModel, DefaultLinkWidget, PointModel } from 'storm-react-diagrams'
+import { AbstractLinkFactory, DefaultLinkModel, DefaultLinkWidget, PointModel, Toolkit } from 'storm-react-diagrams'
+
+import style from './style.scss'
 
 class DeletableLinkWidget extends DefaultLinkWidget {
-  addPointToLink = (event, index: number): void => {
-    if (
-      event.type === 'contextmenu' &&
-      !this.props.diagramEngine.isModelLocked(this.props.link) &&
-      this.props.link.points.length - 1 <= this.props.diagramEngine.getMaxNumberPointsPerLink()
-    ) {
-      event.preventDefault()
-      const point = new PointModel(this.props.link, this.props.diagramEngine.getRelativeMousePoint(event))
-      point.setSelected(true)
-      this.forceUpdate()
-      this.props.link.addPoint(point, index)
-      this.props.pointAdded(point, event)
-    }
+  private currentLink: string
+
+  generatePoint(pointIndex: number): JSX.Element {
+    const { link } = this.props
+    const x = link.points[pointIndex].x
+    const y = link.points[pointIndex].y
+
+    this.currentLink = link.getID()
+
+    return (
+      <g key={`point-${link.points[pointIndex].id}`}>
+        <circle
+          cx={x}
+          cy={y}
+          r={5}
+          className={`point ${this.bem('__point')}${
+            link.points[pointIndex].isSelected() ? this.bem('--point-selected') : ''
+          }`}
+        />
+        <circle
+          onMouseLeave={() => {
+            this.setState({ selected: false })
+          }}
+          onMouseEnter={() => {
+            this.setState({ selected: true })
+          }}
+          data-id={link.points[pointIndex].id}
+          data-linkid={link.id}
+          cx={x}
+          cy={y}
+          r={15}
+          opacity={0}
+          className={`point${this.bem('__point')}`}
+        />
+      </g>
+    )
   }
 
   generateLink(path: string, extraProps: any, id: string | number): JSX.Element {
-    const { link, width, color } = this.props
-    const index = extraProps['data-point'] || 0
+    const { link, diagramEngine } = this.props
+    let { color, width } = this.props
+
+    // @ts-ignore
+    const flowManager = diagramEngine.flowBuilder.manager
+    if (flowManager.shouldHighlightLink(link.getID())) {
+      color = 'var(--ocean)'
```

Comment:
note for all of us : we should have all of our color variables accessible in ui-shared or if we simply override blueprints color set for our variables